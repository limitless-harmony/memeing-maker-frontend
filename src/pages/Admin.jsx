import React from 'react';
import { connect } from 'react-redux';

import Memes from 'components/AdminAllMemes';
import Tagline from 'components/Tagline';
import Title from 'components/Title';

const Home = ({ history }) => (
  <>
    <Tagline />
    <Title />
    <Memes history={history} />
  </>
);

const mapStateToProps = state => ({
  authenticated: state.auth.authenticated,
});

export default connect(mapStateToProps)(Home);
