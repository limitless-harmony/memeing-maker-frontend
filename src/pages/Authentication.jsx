import React, { Component } from 'react';
import { connect } from 'react-redux';
import { parse } from 'query-string';
import { bindActionCreators } from 'redux';

import { login } from 'actions/auth';
import Loading from 'components/Loading';

export class Authentication extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    const { match, location } = props;
    const { code } = parse(location.search);
    const { provider } = match.params;
    const { REACT_APP_API_URL } = process.env;
    if (
      !code &&
      (provider === 'facebook' ||
        provider === 'google' ||
        provider === 'linkedin')
    ) {
      window.location = `${REACT_APP_API_URL}/auth/${provider}`;
    }
  }

  componentDidMount = async () => {
    const {
      actions,
      match,
      previousPath,
      location,
      history,
      authenticated,
    } = this.props;
    const { code } = parse(location.search);
    const { provider } = match.params;
    if (!code) return null;
    const path = previousPath || '/';
    if (authenticated) return history.push(path);
    await actions.login(code, provider);
    return history.push(path);
  };

  render() {
    return <Loading />;
  }
}

const mapStateToProps = state => ({
  previousPath: state.auth.previous,
  authenticated: state.auth.authenticated,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({ login }, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Authentication);
