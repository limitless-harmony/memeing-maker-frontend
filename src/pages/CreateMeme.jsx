import React, { Component, Fragment } from 'react';
import styled from 'styled-components';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import { selectImage } from 'actions/image';
import { showModal } from 'actions/modal';
import Input from 'components/Input';
import { GoArrow } from 'components/Icons';
import MemePreview from 'components/MemePreview';
import { inputFill } from 'styles/colors';
import { calculateRem } from 'styles';
import { getFullImage, composeImage } from 'helpers/image';
import { name } from 'helpers';
import defaultImage from 'assets/images/dank.png';

export class CreateMeme extends Component {
  state = {
    topText: '',
    bottomText: '',
  };

  componentDidMount() {
    this.loadImage();
  }

  selectImage = () => {
    const { actions } = this.props;
    return actions.showModal('select-image');
  };

  loadImage = () => {
    const { actions } = this.props;
    const img = document.createElement('img');
    img.setAttribute('src', `${defaultImage}`);
    img.onload = () => {
      const base64 = getFullImage(img);
      actions.selectImage(base64);
    };
  };

  composeMeme = () => {
    const style = {
      viewBox: '0 0 800 1200',
    };
    composeImage(this.captureDiv, style);
  };

  changeText = event => {
    this.setState({
      [event.currentTarget.name]: event.currentTarget.value,
    });
  };

  render() {
    const { topText, bottomText } = this.state;
    const { selectedImage } = this.props;
    return (
      <Fragment>
        <PreviewContainer
          ref={div => {
            this.captureDiv = div;
          }}
        >
          <MemePreview
            image={selectedImage}
            topText={topText}
            bottomText={bottomText}
            onImageClick={this.selectImage}
          />
        </PreviewContainer>
        <Input
          value={topText}
          name="topText"
          underline
          onChange={this.changeText}
          placeholder="Click to enter top text"
        />
        <Input
          value={bottomText}
          name="bottomText"
          onChange={this.changeText}
          underline
          placeholder="Click to enter bottom text"
        />
        <ButtonContainer>
          Post to
          <SubmitMeme onClick={this.composeMeme}>{name}</SubmitMeme>
          <GoArrow />
        </ButtonContainer>
      </Fragment>
    );
  }
}

export const PreviewContainer = styled.div`
  width: 100%;
`;

export const ButtonContainer = styled.div`
  padding: 0 ${calculateRem(18)};
  margin: ${calculateRem(4)} ${calculateRem(10)};
  display: flex;
  align-items: center;
  box-sizing: border-box;
  width: 100%;
`;

export const SubmitMeme = styled.button`
  color: #4a4a4a;
  background-color: ${inputFill};
  margin: 0 ${calculateRem(14)};
  padding: ${calculateRem(6)} ${calculateRem(13)};
  border: none;
  border-radius: ${calculateRem(10)};
  min-width: 60%;
  text-align: left;
`;

const mapStateToProps = state => ({
  selectedImage: state.image.imageUrl,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({ showModal, selectImage }, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CreateMeme);
