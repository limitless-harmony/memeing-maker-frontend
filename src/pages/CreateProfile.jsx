import React, { Component, Fragment } from 'react';
import styled from 'styled-components';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import { selectImage, showModal, removeImage } from 'actions/common';
import { edit, getOne } from 'actions/meme';
import Input from 'components/Input';
import { GoArrow } from 'components/Icons';
import Preview from 'components/Preview';
import { inputFill } from 'styles/colors';
import { calculateRem } from 'styles';
import { name } from 'helpers';
import defaultImage from 'assets/images/create-meme.svg';

export class CreateProfile extends Component {
  state = {
    topText: '',
    bottomText: '',
    imageUrl: '',
  };

  async componentDidMount() {
    const { match } = this.props;
    const {
      params: { id },
    } = match;
    await this.fetchData(id);
  }

  componentDidUpdate(prevProps) {
    const {
      match: { params },
    } = prevProps;
    const {
      match: {
        params: { id },
      },
    } = this.props;
    if (id !== params.id) {
      this.fetchData(id);
    }
  }

  fetchData = async memeId => {
    const { actions } = this.props;
    await actions.getOne(memeId);
    const { meme, user, history } = this.props;
    if (user.id !== meme.creator.id) {
      return history.push(`/memes/${meme.id}`);
    }
    this.setState(() => {
      return {
        topText: meme.topText,
        bottomText: meme.bottomText,
        imageUrl: meme.image || defaultImage,
      };
    });
    return this.loadImage();
  };

  selectImage = () => {
    const { actions } = this.props;
    return actions.showModal('select-image');
  };

  loadImage = () => {
    const { actions } = this.props;
    const { imageUrl } = this.state;
    actions.selectImage(imageUrl);
  };

  editMeme = async () => {
    const { topText, bottomText } = this.state;
    const { actions, selectedImage, meme } = this.props;
    await actions.edit(
      {
        topText,
        bottomText,
        image: selectedImage,
        name: meme.name,
      },
      meme.id
    );
    actions.removeImage();
  };

  changeText = event => {
    this.setState({
      [event.currentTarget.name]: event.currentTarget.value,
    });
  };

  render() {
    const { topText, bottomText } = this.state;
    const { selectedImage } = this.props;
    return (
      <Fragment>
        <PreviewContainer>
          <Preview
            image={selectedImage}
            topText={topText}
            bottomText={bottomText}
            onImageClick={this.selectImage}
          />
        </PreviewContainer>
        <Input
          value={topText}
          name="topText"
          underline
          onChange={this.changeText}
          placeholder="Enter top text"
        />
        <Input
          value={bottomText}
          name="bottomText"
          onChange={this.changeText}
          underline
          placeholder="Enter bottom text"
        />
        <ButtonContainer>
          Post to
          <Submit onClick={this.editMeme}>{name}</Submit>
          <GoArrow />
        </ButtonContainer>
      </Fragment>
    );
  }
}

export const PreviewContainer = styled.div`
  width: 100%;
`;

export const ButtonContainer = styled.div`
  padding: 0 ${calculateRem(18)};
  margin: ${calculateRem(4)} ${calculateRem(10)};
  display: flex;
  align-items: center;
  box-sizing: border-box;
  width: 100%;
`;

export const Submit = styled.button`
  color: #4a4a4a;
  background-color: ${inputFill};
  margin: 0 ${calculateRem(14)};
  padding: ${calculateRem(6)} ${calculateRem(13)};
  border: none;
  border-radius: ${calculateRem(10)};
  min-width: 60%;
  text-align: left;
`;

const mapStateToProps = state => ({
  selectedImage: state.common.imageUrl,
  meme: state.meme.current,
  user: state.auth.user,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    { showModal, selectImage, edit, getOne, removeImage },
    dispatch
  ),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CreateProfile);
