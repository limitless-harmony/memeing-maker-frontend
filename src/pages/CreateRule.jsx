import React, { Component } from 'react';
import styled from 'styled-components';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';

import {
  selectImage,
  removeImage,
  showModal,
  createRule,
} from 'actions/common';
import defaultImage from 'assets/images/create-meme.svg';
import MemeForm from 'components/MemeForm';

export class CreateRule extends Component {
  state = {
    topText: '',
    bottomText: '',
    size: '',
    defaultImageUrl: defaultImage,
  };

  componentDidMount() {
    this.setState({ size: this.widthRef.offsetWidth });
    this.loadImage();
  }

  selectImage = () => {
    const { actions } = this.props;
    return actions.showModal('select-image');
  };

  loadImage = () => {
    const { actions } = this.props;
    const { defaultImageUrl } = this.state;
    actions.selectImage(defaultImageUrl);
  };

  createMeme = async () => {
    const { topText, bottomText, defaultImageUrl } = this.state;
    const { actions, selectedImage } = this.props;
    if (selectedImage === defaultImageUrl)
      return toast.error('Please select an image first!');

    await actions.createRule({
      topText,
      bottomText,
      image: selectedImage,
    });
    return actions.removeImage();
  };

  changeText = event => {
    this.setState({
      [event.currentTarget.name]: event.currentTarget.value,
    });
  };

  render() {
    const { topText, bottomText, size } = this.state;
    const { selectedImage } = this.props;
    return (
      <Container
        ref={widthRef => {
          this.widthRef = widthRef;
        }}
      >
        <MemeForm
          image={selectedImage}
          topText={topText}
          bottomText={bottomText}
          size={size}
          onChange={this.changeText}
          selectImage={this.selectImage}
          onSubmit={this.createMeme}
          submitText="Rules of play"
        />
      </Container>
    );
  }
}

const Container = styled.div`
  width: 100%;
`;

const mapStateToProps = state => ({
  selectedImage: state.common.imageUrl,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    { showModal, selectImage, removeImage, createRule },
    dispatch
  ),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CreateRule);
