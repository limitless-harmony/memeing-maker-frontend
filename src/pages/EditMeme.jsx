import React, { Component } from 'react';
import styled from 'styled-components';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import { selectImage, showModal, removeImage } from 'actions/common';
import { edit, getOne } from 'actions/meme';
import { inputFill, dark } from 'styles/colors';
import { calculateRem } from 'styles';
import { name } from 'helpers';
import defaultImage from 'assets/images/create-meme.svg';
import MemeForm from 'components/MemeForm';

export class Edit extends Component {
  state = {
    topText: '',
    bottomText: '',
    size: '',
    imageUrl: '',
  };

  componentDidMount() {
    this.setState({ size: this.widthRef.offsetWidth });
    const { match } = this.props;
    const {
      params: { id },
    } = match;
    return this.fetchData(id);
  }

  componentDidUpdate(prevProps) {
    const {
      match: { params },
    } = prevProps;
    const {
      match: {
        params: { id },
      },
    } = this.props;
    if (id !== params.id) {
      this.fetchData(id);
    }
  }

  fetchData = async memeId => {
    const { actions } = this.props;
    await actions.getOne(memeId);
    const { meme, user, history } = this.props;
    if (user.id !== meme.creator.id) {
      return history.push(`/memes/${meme.id}`);
    }
    await this.setState(() => {
      return {
        topText: meme.topText,
        bottomText: meme.bottomText,
        imageUrl: meme.image || defaultImage,
      };
    });
    return this.loadImage();
  };

  selectImage = () => {
    const { actions } = this.props;
    return actions.showModal('select-image');
  };

  loadImage = () => {
    const { actions } = this.props;
    const { imageUrl } = this.state;
    actions.selectImage(imageUrl);
  };

  editMeme = async () => {
    const { topText, bottomText } = this.state;
    const { actions, selectedImage, meme } = this.props;
    await actions.edit(
      {
        topText,
        bottomText,
        image: selectedImage,
        name: meme.name,
      },
      meme.id
    );
    actions.removeImage();
  };

  changeText = event => {
    this.setState({
      [event.currentTarget.name]: event.currentTarget.value,
    });
  };

  render() {
    const { topText, bottomText, size } = this.state;
    const { selectedImage } = this.props;
    return (
      <Container
        ref={widthRef => {
          this.widthRef = widthRef;
        }}
      >
        <MemeForm
          image={selectedImage}
          topText={topText}
          bottomText={bottomText}
          size={size}
          onChange={this.changeText}
          selectImage={this.selectImage}
          onSubmit={this.editMeme}
          submitText={name}
        />
      </Container>
    );
  }
}

export const Container = styled.div`
  width: 100%;
`;

export const ButtonContainer = styled.div`
  padding: 0 ${calculateRem(18)};
  margin: ${calculateRem(4)} ${calculateRem(10)};
  display: flex;
  align-items: center;
  box-sizing: border-box;
  width: 100%;
`;

export const SubmitMeme = styled.button`
  color: ${dark};
  background-color: ${inputFill};
  margin: 0 ${calculateRem(14)};
  padding: ${calculateRem(6)} ${calculateRem(13)};
  border: none;
  cursor: pointer;
  border-radius: ${calculateRem(10)};
  min-width: 60%;
  text-align: left;
`;

const mapStateToProps = state => ({
  selectedImage: state.common.imageUrl,
  meme: state.meme.current,
  user: state.auth.user,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    { showModal, selectImage, edit, getOne, removeImage },
    dispatch
  ),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Edit);
