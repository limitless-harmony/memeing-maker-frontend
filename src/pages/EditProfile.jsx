import React, { Component } from 'react';
import styled from 'styled-components';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import { selectImage, showModal, removeImage } from 'actions/common';
import { edit, savePathFrom, clearPathFrom } from 'actions/auth';
import Input from 'components/Input';
import { pink } from 'styles/colors';
import { calculateRem } from 'styles';
import defaultImage from 'assets/images/serious-cat.jpg';
import MemeForm from 'components/MemeForm';

export class EditProfile extends Component {
  state = {
    topText: '',
    bottomText: '',
    image: '',
    username: '',
    error: '',
    size: '',
  };

  async componentDidMount() {
    this.setState({ size: this.widthRef.offsetWidth });
    const { authenticated, history, actions } = this.props;
    if (!authenticated) {
      const { pathname } = history.location;
      actions.savePathFrom(pathname);
      return history.push('/login');
    }
    return this.loadData();
  }

  loadData = async () => {
    const { user } = this.props;
    const { topText, bottomText, username, image } = user;
    await this.setState(() => {
      return {
        topText,
        bottomText,
        image: image || defaultImage,
        username,
      };
    });
    return this.loadImage();
  };

  selectImage = () => {
    const { actions } = this.props;
    return actions.showModal('select-image');
  };

  loadImage = () => {
    const { actions } = this.props;
    const { image } = this.state;
    actions.selectImage(image);
  };

  editProfile = async () => {
    const { topText, bottomText, username } = this.state;
    const { actions, selectedImage, history, previousPath } = this.props;
    if (!username)
      return this.setState({ error: 'Please choose a unique username!' });
    await actions.edit({
      topText,
      bottomText,
      image: selectedImage,
      username,
    });
    this.loadData();
    actions.removeImage();
    const path = previousPath || '/';
    history.push(path);
    return actions.clearPathFrom();
  };

  changeText = event => {
    if (event.currentTarget.name === 'username') this.setState({ error: '' });
    this.setState({
      [event.currentTarget.name]: event.currentTarget.value,
    });
  };

  render() {
    const { topText, bottomText, username, error, size } = this.state;
    const { selectedImage } = this.props;
    return (
      <>
        <Container
          ref={widthRef => {
            this.widthRef = widthRef;
          }}
        >
          <Input
            value={username}
            name="username"
            underline
            focused
            onChange={this.changeText}
            placeholder="Choose a unique username"
          />
          {error && <Error>{error}</Error>}
          <MemeForm
            image={selectedImage}
            topText={topText}
            bottomText={bottomText}
            size={size}
            onChange={this.changeText}
            selectImage={this.selectImage}
            onSubmit={this.editProfile}
            submitText="Player Profile"
          />
        </Container>
      </>
    );
  }
}

const Container = styled.div`
  width: 100%;
`;

const Error = styled.div`
  width: 100%;
  font-size: ${calculateRem(13)};
  color: ${pink};
  text-align: center;
  margin: 0 auto;
`;

const mapStateToProps = state => ({
  selectedImage: state.common.imageUrl,
  authenticated: state.auth.authenticated,
  previousPath: state.auth.previous,
  user: state.auth.user,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    { showModal, selectImage, edit, removeImage, savePathFrom, clearPathFrom },
    dispatch
  ),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditProfile);
