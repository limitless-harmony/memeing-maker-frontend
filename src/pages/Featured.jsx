import React, { Component } from 'react';
import styled from 'styled-components';

import { calculateRem } from 'styles';
import Masonry from 'components/Masonry';
import MemeCard from 'components/MemeCard';
import Button from 'components/Button';
import Yes from 'assets/images/yes.png';
import Sense from 'assets/images/sense.jpg';
import Duh from 'assets/images/duh.jpg';
import { buttonBorder, black } from 'styles/colors';

class Featured extends Component {
  greet = () => {};

  render() {
    return (
      <StyledFeatured>
        <Slogan>Make meaning. Share meaning.</Slogan>
        <Title>Memeing Maker</Title>
        <SectionHeading>This Weekâ€™s Featured Memes</SectionHeading>
        <Masonry>
          <MemeCard
            imageUrl="https://loremflickr.com/240/320"
            topText="Randomly generated"
            bottomText="New every time"
          />
          <MemeCard
            imageUrl={Yes}
            topText="A top text"
            bottomText="A bottom text"
          />
          <MemeCard
            imageUrl={Sense}
            topText="When you know"
            bottomText="And you know that you know"
          />
          <MemeCard
            imageUrl={Duh}
            topText="You think you know me"
            bottomText="Duh! You have no idea. I am like the wind!"
          />
          <MemeCard
            imageUrl="https://loremflickr.com/320/240"
            topText="You think you know me"
            bottomText="Duh! You have no idea. I am like the wind!"
          />
          <MemeCard
            imageUrl="https://loremflickr.com/240/320"
            topText="Randomly generated"
            bottomText="New every time"
          />
          <MemeCard
            imageUrl="https://loremflickr.com/240/320"
            topText="Randomly generated"
            bottomText="New every time"
          />
        </Masonry>
        <Button color={buttonBorder} outline rounded>
          <ButtonText>More</ButtonText>
        </Button>
      </StyledFeatured>
    );
  }
}

const StyledFeatured = styled.div`
  margin: ${calculateRem(40)} auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  text-align: center;
`;

export const Title = styled.div`
  margin: 0 auto ${calculateRem(60)};
  line-height: ${calculateRem(43)};
  font-size: ${calculateRem(36)};
`;

export const Slogan = styled.div`
  margin: 0 auto;
`;
export const SectionHeading = styled.div`
  font-size: ${calculateRem(18)};
  margin: ${calculateRem(16)} auto;
`;

const ButtonText = styled.div`
  color: ${black};
`;

export default Featured;
