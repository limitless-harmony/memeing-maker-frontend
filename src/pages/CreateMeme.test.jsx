import React from 'react';
import { shallow, mount } from 'enzyme';

import { CreateMeme } from 'pages/CreateMeme';

import * as imageHelper from 'helpers/image';

const mockFunction = jest.fn();

describe('Testing the CreateMeme page component', () => {
  const props = {
    selectedImage: '/images/me.png',
    actions: {
      showModal: mockFunction,
      selectImage: mockFunction,
    },
  };
  it('renders without crashing', () => {
    const wrapper = shallow(<CreateMeme />);
    expect(wrapper.find('div'));
    expect(wrapper).toMatchSnapshot();
  });
  it('it loads an image when the component mounts', () => {
    const wrapper = shallow(<CreateMeme {...props} />);
    const instance = wrapper.instance();
    const spy = jest.spyOn(instance, 'loadImage');
    instance.componentDidMount();
    expect(spy).toHaveBeenCalled();
  });
  it('invoking changeText updates the state', () => {
    const wrapper = shallow(<CreateMeme {...props} />);
    const instance = wrapper.instance();
    expect(wrapper.state('topText')).toBe('');
    const event = {
      currentTarget: {
        name: 'topText',
        value: 'Veraclins',
      },
    };
    instance.changeText(event);
    expect(wrapper.state('topText')).toBe('Veraclins');
  });
  it('invoking selectImage opens the select-image modal', () => {
    const wrapper = shallow(<CreateMeme {...props} />);
    const instance = wrapper.instance();
    instance.selectImage();
    expect(mockFunction).toHaveBeenCalled();
  });
  it('it composes the meme into an image', () => {
    imageHelper.composeImage = mockFunction;
    const wrapper = mount(<CreateMeme {...props} />);
    const instance = wrapper.instance();
    instance.composeMeme();
    expect(mockFunction).toHaveBeenCalled();
  });
});
