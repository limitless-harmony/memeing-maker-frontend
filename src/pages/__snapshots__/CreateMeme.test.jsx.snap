// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the CreateMeme page component renders without crashing 1`] = `
<Fragment>
  <styled.div>
    <MemePreview
      bottomText=""
      onImageClick={[Function]}
      topText=""
    />
  </styled.div>
  <Input
    name="topText"
    onChange={[Function]}
    placeholder="Click to enter top text"
    underline={true}
    value=""
  />
  <Input
    name="bottomText"
    onChange={[Function]}
    placeholder="Click to enter bottom text"
    underline={true}
    value=""
  />
  <styled.div>
    Post to
    <styled.button
      onClick={[Function]}
    >
      Memeing Maker
    </styled.button>
    <ForwardRef />
  </styled.div>
</Fragment>
`;
